use-cases/xtclim/hpo.py

```diff
@@ -0,0 +1,259 @@
+import argparse
+import os
+import sys
+from pathlib import Path
+from typing import Dict
+import yaml
+
+import matplotlib.pyplot as plt
+import ray
+import torch
+from ray import train, tune
+
+from itwinai.parser import ConfigParser
+
+sys.path.append(os.path.join(os.path.dirname(__file__), 'src'))
+sys.path.append(os.path.join(os.path.dirname(__file__), 'preprocessing'))
+
+
+def run_trial(config: Dict, data: Dict):
```

Comment:
here it seems to me that, for each hyperparam config, the pipeline is run for all seasons. As far as I understood, the preprocessing steps in the pipeline are doing always the same operations, resulting in saving some dataset to disk.
It may be more efficient to generate the datasets that stay constant across seasons and HPO trials only once. This can be done by executing only a subset of steps _before_ even starting HPO, and executing during HPO only the steps that need to be tuned.

This can be done by slicing the pipeline before executing it:

```python
# Only preproc
my_pipeline = parser.parse_pipeline()
preproc_steps = my_pipeline[:2] # keep only first 2 steps
preproc_steps.execute(...)

# Only training
my_pipeline = parser.parse_pipeline()
train_steps = my_pipeline[2:] # keep only from third step on
train_steps .execute(...)
```